app/models/profile.rb

```diff
@@ -22,6 +26,7 @@ class Profile < ApplicationRecord
   validates :account, presence: true, if: -> { hosts.any? }
 
   after_update :destroy_orphaned_business_objective
+  before_destroy :destroy_all_test_results, if: -> { delete_all_test_results }
```

Comment:
Why setting an attribute instead of manually calling `profile.test_results.destroy_all` and submitting the job I mention below as part of the mutation?

Reply:
The reason is that it doesn't feel like the responsibility of the mutation to remove a profiles test results.
 It shouldn't even know about the `delete_all_test_results` attribute, actually. At the moment it just passes it on, so its ...fine. :)

The way I see it, the mutation and GraphQL endpoint is the mediatior between User intput and Business Logic. And so, to me it's ok for it to know how to delete a `Profile`, but it shouldn't know how to delete their `TestResult`s, that is only for the Profile to know.